stan/math/rev/fun/lambert_w.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef STAN_MATH_REV_FUN_LAMBERT_W_HPP
+#define STAN_MATH_REV_FUN_LAMBERT_W_HPP
+
+#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/meta.hpp>
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/fun/boost_policy.hpp>
+#include <stan/math/prim/fun/lambert_w.hpp>
+
+namespace stan {
+namespace math {
+
+namespace internal {
+class lambertw0_vari : public op_v_vari {
+ public:
+  explicit lambertw0_vari(vari* avi) : op_v_vari(lambert_w0(avi->val_), avi) {}
+  void chain() {
+    if (avi_->val_ == 0.0) {
```

Comment:
Does the else case not work here?  If it does, then remove the conditional and just use the general case.  Conditionals are very expensive when speculative evaluation fails and the CPU has to backtrack.  It's better to pay a penalty most of the time in doing a little more arithmetic in the mainline case.  It also helps with inlining.

Reply:
As far that I understand @SteveBronder added the clause because it leads to failures otherwise. Maybe he can enlighten us

Reply:
I can check tmrw but I remember it failing without the else clause

Reply:
Nope that if can be deleted!